core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -151,6 +168,28 @@ class SubmissionSupervisor(executionServiceCluster: ExecutionServiceCluster,
     }
   }
 
+  private def startMonitoringNewSubmissions: Future[SubmissionMonitorPassComplete.type] = {
+    datasource.inTransaction { dataAccess =>
+      dataAccess.submissionQuery.listAllActiveSubmissions() map { activeSubs =>
+        val monitoredSubmissions = context.children.map(_.path.name).toSet
```

Comment:
I don't think you should be closing over `context` inside a future. It's mutable state within the actor, so it should only be accessed by the actor itself. The akka docs warn against this:
```   
* WARNING: Only valid within the Actor itself, so do not close over it and
* publish it to other threads!
```

It would be better to compute `monitoredSubmissions` outside of the database transaction (it doesn't need the DB results anyway).

(Note: `self` is immutable so doing `self ! msg` inside a future is fine.)